src/Core/src/Platform/Windows/StackNavigationManager.cs

```diff
@@ -156,31 +154,41 @@ void OnNavigated(object sender, UI.Xaml.Navigation.NavigationEventArgs e)
 				return;
 
 			var nv = NavigationView;
-			ContentPresenter? presenter;
+			ContentControl? presenter;
 
 			if (page.Content == null)
 			{
-				presenter = new ContentPresenter()
+				presenter = new ContentControl()
 				{
 					HorizontalAlignment = UI.Xaml.HorizontalAlignment.Stretch,
 					VerticalAlignment = UI.Xaml.VerticalAlignment.Stretch
 				};
 
+				// It's possible that the unloaded event didn't happen yet, so force clear the content
```

Comment:
Note: we don't just run this by default because when the transition is animated the page flashes blank.